rpc-server/src/modules/queries/utils.rs

```diff
@@ -115,16 +119,36 @@ pub async fn fetch_state_from_db(
         prefix,
     );
     let state_from_db = get_state_keys_from_db(db_manager, account_id, block_height, prefix).await;
-    if state_from_db.is_empty() {
+    if state_from_db.is_empty() && optimistic_data.is_empty() {
```

Comment:
I wouldn't say I like the idea of putting this logic here. I agree to have it here for now but consider adding a TODO to refactor it later.